lint/util.py

```diff
@@ -610,3 +586,45 @@ def load_json(*segments, from_sl_dir=False):
     base_path = "Packages/SublimeLinter/" if from_sl_dir else ""
     full_path = base_path + "/".join(segments)
     return sublime.decode_value(sublime.load_resource(full_path))
+
+
+# syntax functions
+
+
+def yield_tail_scopes(scope):
```

Comment:
This function yields the following valus over `text.html.markdown`:
```
"markdown"
"html"
"source"
```

Shouldn't you be eliminating the tail and yield `scope` here? So the result becomes:
```
"source.html.markdown"
"source.html"
"source"
```